pallets/msa/src/lib.rs

```diff
@@ -408,22 +409,23 @@ pub mod pallet {
 		///
 		/// ### Errors
 		/// - Returns [`InvalidSelfRevoke`](Error::InvalidSelfRevoke) if `origin` and `key` are the same.
-		/// - Returns [`KeyRevoked`](Error::KeyRevoked) if `key` is already expired.
 		/// - Returns [`NotKeyOwner`](Error::NotKeyOwner) if `origin` does not own the MSA ID associated with `key`.
 		/// - Returns [`NotKeyExists`](Error::NoKeyExists) if `origin` or `key` are not associated with `origin`'s MSA ID.
-		///
-		#[pallet::weight(T::WeightInfo::revoke_msa_key())]
-		pub fn revoke_msa_key(origin: OriginFor<T>, key: T::AccountId) -> DispatchResult {
+		/// ### Remarks
+		/// - Removal of key deletes the association of the key with the MSA.
+		/// - The key can be re-added to same or another MSA if needed.
+		#[pallet::weight(T::WeightInfo::delete_msa_key())]
+		pub fn delete_msa_key(origin: OriginFor<T>, key: T::AccountId) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
 			ensure!(who != key, Error::<T>::InvalidSelfRevoke);
 
 			let who = Self::try_get_key_info(&who)?;
 			let key_info = Self::try_get_key_info(&key)?;
-			ensure!(who.expired == T::BlockNumber::zero(), Error::<T>::KeyRevoked);
 			ensure!(who.msa_id == key_info.msa_id, Error::<T>::NotKeyOwner);
 
-			Self::revoke_key(&key)?;
+			Self::delete_key_for_msa(who.msa_id, &key)?;
+			Self::delete_key_info(&key)?;
```

Comment:
we are not deleting the key_info we are deleting the key.  I suggest not making two functions for this and instead combining them into one here:
```
Self::delete_key(&key)?;
```

Either we make this transactional or we combine them. Lets pair.



delete_key_info is not only deleting the key but the

Reply:
Currently yes delete_key_for_msa deletes the association of key with msa from MSAKeysOf storage while delete_key_info deletes the keyinfo of that key from KeyInfo storage map

We can pair for sure

Reply:
@enddynayn  I combined the two in one where key is deleted from MSAKeys as well as KeyInfo

lets talk about its transactional nature

Reply:
@enddynayn  maybe I am mistaken this function is within a transaction (extrinsic) if otherwise you mean something else